homeassistant/components/nobo_hub/config_flow.py

```diff
@@ -0,0 +1,254 @@
+
+    def __init__(self, msg) -> None:
+        """Instantiate error."""
+        super().__init__()
+        self.msg = msg
+
+
+class OptionsFlowHandler(config_entries.OptionsFlow):
+    """Handles options flow for the component."""
+
+    def __init__(self, config_entry: config_entries.ConfigEntry) -> None:
+        """Initialize the options flow."""
+        self.config_entry = config_entry
+
+    async def async_step_init(self, user_input=None) -> FlowResult:
+        """Manage the options."""
+
+        hub = self.hass.data[DOMAIN][self.config_entry.entry_id]
+        profile_names = sorted(
+            k["name"].replace("\xa0", " ") for k in hub.week_profiles.values()
+        )
+
+        if user_input is not None:
+            off_command = user_input.get(CONF_COMMAND_OFF)
+            if off_command not in profile_names:
+                off_command = None
+            on_commands = {}
+            for key, on_command in user_input.items():
+                if key.startswith(CONF_COMMAND_ON + "_zone_"):
+                    zone = key[16:]
+                    if on_command not in profile_names:
+                        on_command = None
+                    on_commands[
+                        hub.zones[zone]["name"].replace("\xa0", " ")
+                    ] = on_command
+
+            data = {
+                CONF_OVERRIDE_TYPE: user_input.get(CONF_OVERRIDE_TYPE),
+                CONF_COMMAND_OFF: off_command,
+                CONF_COMMAND_ON: on_commands,
+            }
+
+            return self.async_create_entry(title="", data=data)
+
+        override_type = self.config_entry.options.get(CONF_OVERRIDE_TYPE)
+        if override_type != CONF_OVERRIDE_TYPE_NOW:
+            override_type = CONF_OVERRIDE_TYPE_CONSTANT
+
+        profile_names.insert(0, CONF_WEEK_PROFILE_NONE)
+        profiles = vol.Schema(vol.In(profile_names))
+
+        off_command = self.config_entry.options.get(CONF_COMMAND_OFF)
+        if off_command not in profile_names:
+            off_command = CONF_WEEK_PROFILE_NONE
+        schema = vol.Schema(
+            {
+                vol.Required(CONF_OVERRIDE_TYPE, default=override_type): vol.In(
+                    [CONF_OVERRIDE_TYPE_CONSTANT, CONF_OVERRIDE_TYPE_NOW]
+                ),
+                # Ideally we should use vol.Optional, but resetting the field in the UI
+                # will default to the old value instead of setting to None.
+                vol.Required(CONF_COMMAND_OFF, default=off_command): profiles,
+            }
+        )
+
+        placeholder = ""
+        on_commands = self.config_entry.options.get(CONF_COMMAND_ON)  # type: ignore[assignment]
+        if on_commands is None:
+            on_commands = {}
+        for zone in hub.zones:
+            name = hub.zones[zone]["name"].replace("\xa0", " ")
+            on_command = (
+                on_commands[name]
+                if name in on_commands and on_commands[name] in profile_names
```

Comment:
Please don't use ternary operator expression if it spans more than one line.

Reply:
Fixed.